### ISSUE TITLE
base_url set in parent class doesn't work in child class

### ISSUE THREAD

**Issue by adisos** [andyjeffries/flexirest#4]
I have to set manually on inherited class to make it working.

**Comment by andyjeffries** [comment]
Can you give a code sample that uses http://mockbin.org/ or something.

**Comment by adisos** [comment]
```
class MyProject::Base < Flexirest::Base
  base_url Rails.application.config.api_server_url
  before_request do |name, request|
    request.get_params[:api_key] = "1234567890-1234567890"
  end
end

class Person < MyProject::Base
  # No need to declare a before_request for :api_key, already defined by the parent
end
```

**Comment by adisos** [comment]
Look, I add `base_url` to parent class and when I do `Person.find(1)` it shows:
```
[1] pry(main)> Person.find(1)
  Flexirest Person:/people/1?api_key=1234567890-1234567890 - Trying to read from cache
  Flexirest Person#find - Etag cached copy found with etag W/"773b83a5e16736e02403ff961cba546e"
  Flexirest (1.5ms) Person#find
URI::InvalidURIError: bad URI(is not URI?): 
from /home/user/.rvm/rubies/ruby-2.2.1/lib/ruby/2.2.0/uri/rfc3986_parser.rb:17:in `rescue in split'
```

**Comment by andyjeffries** [comment]
You're right, this should be expected behaviour.  Fixed in v1.2.3.

**Comment by adisos** [comment]
Thanks!
